src/Model/ItemId.php

```diff
@@ -23,7 +24,9 @@ public function __toString() : string
 
     public static function fromString(string $id) : ItemId
     {
-        if (!preg_match('/^([A-Za-z0-9-._~!$&\'()*+,;=:@]|%[A-F0-9]{2})+$/', $id)) {
```

Comment:
in a perfect world `preg_match` would return a boolean, but since it's a wrapper over the PCRE C library; there are many examples of this in PHP. But the original idiom ("if it doesn't match") captures the intent better than the strict type version (is it the number of matches? Or a boolean-as-int result? An additional variable is added to the local scope despite only being needed for the condition)

Reply:
Did look at using [Safe PHP](https://github.com/thecodingmachine/safe), but seems a bit unstable/looks like it could potentially have PHP version problems (re changing signatures).

Only way to avoid the variable is to use a self-executing function/wrap it locally.